diff --git a/CMakeLists.txt b/CMakeLists.txt
index f722f35ad1be..d910c1a46045 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,6 +39,11 @@ if (NOT XPTI_PROXY_SRC)
   set(XPTI_PROXY_SRC ${CMAKE_CURRENT_SOURCE_DIR}/../xpti/src/xpti_proxy.cpp)
 endif()
 
+find_package(LLVM REQUIRED CONFIG)
+include(AddLLVM)
+
+find_package(Clang REQUIRED CONFIG)
+
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules")
 include(AddSYCLExecutable)
 include(AddSYCL)
@@ -142,13 +147,14 @@ endif()
 set(CLANG_VERSION "${CLANG_VERSION_MAJOR}.${CLANG_VERSION_MINOR}.${CLANG_VERSION_PATCHLEVEL}")
 
 set(SYCL_INCLUDE_DIR "include")
-set(SYCL_INCLUDE_BUILD_DIR ${LLVM_BINARY_DIR}/${SYCL_INCLUDE_DIR})
+set(SYCL_INCLUDE_BUILD_DIR ${CMAKE_BINARY_DIR}/${SYCL_INCLUDE_DIR})
+include_directories(${SYCL_INCLUDE_BUILD_DIR})
 
-add_llvm_external_project(opencl)
-list(FIND LLVM_ENABLE_PROJECTS opencl OPENCL_PROJ_FOUND)
-if(OPENCL_PROJ_FOUND EQUAL -1)
-  message(FATAL_ERROR "opencl external project required but not found.")
-endif()
+# TODO: I'm unsure whether this will just link to regular OpenCL since both are present
+find_package(OpenCL REQUIRED)
+message(OpenCL_INCLUDE_DIR)
+# Ensure we actually have opencl-aot and not just regular OpenCL
+find_program(OPENCL_AOT_EXECUTABLE NAMES opencl-aot REQUIRED)
 
 if(NOT OpenCL_FOUND)
   # Copy OpenCL Headers into sycl headers build directory
@@ -372,7 +377,6 @@ add_custom_target(sycl-compiler
           llvm-objcopy
           spirv-to-ir-wrapper
           sycl-post-link
-          opencl-aot
 )
 
 add_custom_target( sycl-runtime-libraries
