diff --git a/llvm/tools/llvm-config/llvm-config.cpp b/llvm/tools/llvm-config/llvm-config.cpp
index c7cb05b82821..cf59533b6b75 100644
--- a/llvm/tools/llvm-config/llvm-config.cpp
+++ b/llvm/tools/llvm-config/llvm-config.cpp
@@ -365,10 +365,10 @@ int main(int argc, char **argv) {
       sys::fs::make_absolute(ActivePrefix, Path);
       ActiveBinDir = std::string(Path);
     }
-    {
-      SmallString<256> Path(LLVM_INSTALL_LIBDIR LLVM_LIBDIR_SUFFIX);
-      sys::fs::make_absolute(ActivePrefix, Path);
-      ActiveLibDir = std::string(Path);
+    if (StringRef(LLVM_INSTALL_LIBDIR).startswith("/")) {
+      ActiveLibDir = LLVM_INSTALL_LIBDIR;
+    } else {
+      ActiveLibDir = ActivePrefix + "/" + LLVM_INSTALL_LIBDIR;
     }
     {
       SmallString<256> Path(LLVM_INSTALL_PACKAGE_DIR);
